site/content/posts/2021-11-09-E2E-Skeleton-Plugin.md

```diff
@@ -0,0 +1,135 @@
+ - Creates/destroys namespaces for each test to avoid collisions
+ - Sends progress updates to the Sonobuoy aggregator so that `sonobuoy status` can report on test progress while running
+
+In this blog post I'll show off the plugin, its features, and show you how to skip the boilerplate so you can get to testing your business logic.
+
+## How to use the e2e-skeleton plugin
+
+This is a "plugin" which is _meant_ to be modified prior to use. Currently, it is
+just a skeleton for you to put your tests into. So in order to get set up and write your own tests,
+you have to:
+
+ - Clone the repo (or copy the example code) `git clone https://github.com/vmware-tanzu/sonobuoy-plugins`
+ - Modify `build.sh` to contain your own registry/image name
+ - Modify `plugin.yaml` to reference the registry/image for your tests
+ - Add more tests!
+
+## Integrations with Sonobuoy
+
+#### Preconfigured Project Files
+ 
+Since this code is meant to be your starting point for your own plugin, we
+include the basic files you'll need during your plugin's lifecycle:
+
+- Dockerfile
+  - All Sonobuoy plugins end up launching containers in your Kubernetes cluster. So to make a plugin you'll need to containerize your code. We've done that for you already and used a base image that was easy to work with and debug. Feel free to modify this as desired.
+- Instructions/script to build your code
+  - You've got the code and Dockerfile, but you still have to actually build/push the container. We provide a simple `build.sh` script that will build/push the image to your specified registry. (This is currently a very minimal starting point and would love help making it more robust.)
+- The plugin.yaml definition
+   - This is the file that Sonobuoy reads to understand your plugin and what to execute. As you create your plugin, you'll just need to tweak this file to reference your container registry/image and any other small changes you may want to add such as providing unique flags to the test invocation.
+ 
+Once you copy this plugin code locally and write your tests, all you'll need to do to build and run your plugin is:
+```bash
+$ ./build.sh
+$ sonobuoy run -p ./plugin.yaml --wait
+``` 
+
+#### Reporting Format
+
+The test output is JSON produced via `go test` which Sonobuoy
+parses to provide useful status/results reports via `sonobuoy status` and `sonobuoy results`.
+ 
+When you use this plugin you won't have to worry about the
+test output format at all or think about which testing library to import. Just write your tests.
+
+#### Progress updates
+
+This plugin skeleton will provide progress updates back to the Sonobuoy aggregator
+so that users waiting on results can see not only that tests are still running, but
+how many have passed/failed/completed (and what those failures are).
+
+To get basic progress information, just run:
+```bash
+$ sonobuoy status
+```
+
+To get the complete status information, add the `--json` flag.
+
+## Benefits of E2E Framework
```

Comment:
You should mention above that you are using the e2e-framework for this plugin (at least I didn't see that).

Reply:
Oh ok, thanks I'll reread the intro and such and make sure its there. I know I put this section and then mentioned it in the outro but may have lost those bullets when working from my outline.